spec/controllers/recipes_controller_spec.rb

```diff
@@ -0,0 +1,166 @@
+require 'rails_helper'
+
+describe RecipesController do
+  let(:valid_attributes) { attributes_for(:recipe) }
+
+  context 'user is not signed in' do
+    describe 'GET index' do
+      it 'exposes all recipes' do
+        get :index
+        expect(response).to be_success
+        expect(response).to render_template(:index)
+      end
+    end
+
+    describe 'GET show' do
+      let(:recipe) { build(:recipe) }
+      before {
+        expect(Recipe).to receive(:find).with('1').and_return(recipe)
+      }
+      it 'exposes the requested book' do
+        get :show, params: { id: 1 }
+        expect(response).to be_success
+        expect(response).to render_template(:show)
+      end
+    end
+
+    describe 'GET new' do
+      it 'redirects user to the login page' do
+        get :new
+        expect(response).not_to be_success
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
+      end
+    end
+
+    describe 'POST create' do
+      it 'redirects user to the login page' do
+        post :create, recipe: valid_attributes
+        expect(response).not_to be_success
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
+      end
+    end
+
+    describe 'GET edit' do
+      let(:recipe) { build(:recipe) }
+      before {
+        expect(Recipe).not_to receive(:find).with('1')
+      }
+      it 'redirects user to the login page' do
+        get :edit, params: { id: 1 }
+        expect(response).not_to be_success
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
+      end
+    end
+
+    describe 'PUT update' do
+      let(:recipe) { build(:recipe) }
+      before {
+        expect(Recipe).not_to receive(:find).with('1')
+      }
+      it 'redirects user to the login page' do
+        put :update, params: { id: 1 }, recipe: valid_attributes
+        expect(response).not_to be_success
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
+      end
+    end
+
+    describe 'DELETE destroy' do
+      let(:recipe) { build(:recipe) }
+      before {
+        expect(Recipe).not_to receive(:find).with('1')
+      }
+      it 'redirect user to the login page' do
+        delete :destroy, params: { id: 1 }
+        expect(response).not_to be_success
+        expect(response).to redirect_to(new_user_session_path)
+        expect(flash[:alert]).to eq('You need to sign in or sign up before continuing.')
+      end
+    end
+  end
+
+  context 'user is signed in' do
+    login_user
+    before :each do
```

Comment:
to nie jest błąd, ale możesz pomijać `:each` - domyślnie samo 
```
before {
...
}
```
Zachowa się tak samo jak `before :each`